packages/round-manager/src/features/round/__tests__/ViewFundingAdmin.test.tsx

```diff
@@ -0,0 +1,346 @@
+    fireEvent.click(fundingAdminTab);
+    expect(screen.getByTestId("match-stats-title")).toBeInTheDocument();
+    expect(screen.getByTestId("matching-stats-table")).toBeInTheDocument();
+    expect(screen.getByTestId("custom-or-default-test-id")).toBeInTheDocument();
+  
+    const customRadioButton = screen.getByTestId("custom-radio-test-id");
+    fireEvent.click(customRadioButton);
+    expect(screen.getByTestId("dropzone")).toBeInTheDocument();
+  
+    // test dropzone .json file upload
+    const dropzone = screen.getByTestId("dropzone");
+    fireEvent.drop(dropzone, { dataTransfer: { files: [mockFile] } });
+  
+    expect(mockFile.arrayBuffer).toHaveBeenCalled();
+    await waitFor(() => {
+      expect(screen.getByTestId("matching-stats-table")).toBeInTheDocument();
+    });
+  });
+  
+
+  it("does not upload an invalid json file when dropped in dropzone", async () => {
+    (useRoundMatchData as jest.Mock).mockImplementation(() => ({
+      return: {
+        data: [makeQFDistribution(), makeQFDistribution()],
+        error: null,
+        loading: false,
+      },
+    }));
+  
+    // mock file.arrayBuffer
+    const mockFile = new File(
+      ["{}"],
+      "test.json", { type: "application/json" }
+    );
+    const mockFileArrayBuffer = jest.fn().mockResolvedValue(new ArrayBuffer(0));
+    Object.defineProperty(mockFile, "arrayBuffer", {
+      value: mockFileArrayBuffer,
+    });
+
+    const roundEndTime = faker.date.recent();
+    const roundStartTime = faker.date.past(1, roundEndTime);
+    const applicationsEndTime = faker.date.past(1, roundStartTime);
+    const applicationsStartTime = faker.date.past(1, applicationsEndTime);
+  
+    const approvedProjects = [
+      makeApprovedProjectData(),
+      makeApprovedProjectData(),
+      makeApprovedProjectData(),
+    ];
+    mockRoundData = makeRoundData({
+      applicationsStartTime,
+      applicationsEndTime,
+      roundStartTime,
+      roundEndTime,
+      approvedProjects,
+    });
+    render(
+      wrapWithBulkUpdateGrantApplicationContext(
+        wrapWithApplicationContext(
+          wrapWithReadProgramContext(
+            wrapWithRoundContext(<ViewRoundPage />, {
+              data: [mockRoundData],
+              fetchRoundStatus: ProgressStatus.IS_SUCCESS,
+            })
+          )
+        )
+      )
+    );
+    const fundingAdminTab = screen.getByTestId("funding-admin");
+    fireEvent.click(fundingAdminTab);
+    expect(screen.getByTestId("match-stats-title")).toBeInTheDocument();
+    expect(screen.getByTestId("matching-stats-table")).toBeInTheDocument();
+    expect(screen.getByTestId("custom-or-default-test-id")).toBeInTheDocument();
+  
+    const customRadioButton = screen.getByTestId("custom-radio-test-id");
+    fireEvent.click(customRadioButton);
+    expect(screen.getByTestId("dropzone")).toBeInTheDocument();
+
+    // test dropzone .json file upload
+    const dropzone = screen.getByTestId("dropzone");
```

Comment:
TODO: this piece fails. I think it's how we've set the mockFile but I'm not sure

Reply:
resolved by mocking the text decoder